src/update.rs

```diff
@@ -494,6 +494,7 @@ fallback_impl! { compact_str::CompactString, }
 
 macro_rules! tuple_impl {
     ($($t:ident),*; $($u:ident),*) => {
+        #[diagnostic::do_not_recommend]
```

Comment:
This also seemed like a nice quality of life improvement. The error message just lists all the tuple types otherwise.

Reply:
You can bring this back, we bumped the MSRV now (needs a rebase then)